front/pages/api/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/content.ts

```diff
@@ -0,0 +1,115 @@
+import type {
+  ContentNodesViewType,
+  GetDataSourceOrViewContentResponseBody,
+  WithAPIErrorResponse,
+} from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { getDataSourceContent } from "@app/lib/api/vaults";
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { DataSourceViewResource } from "@app/lib/resources/data_source_view_resource";
+import { apiError } from "@app/logger/withlogging";
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorResponse<GetDataSourceOrViewContentResponseBody>
+  >,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const dataSourceView = await DataSourceViewResource.fetchById(
+    auth,
+    req.query.dsvId as string
+  );
+
+  const dataSource = dataSourceView?.dataSource;
+  const vault = dataSourceView?.vault;
+
+  if (
+    !dataSourceView ||
+    !dataSource ||
+    !vault ||
+    req.query.vId !== vault.sId ||
+    (!auth.isAdmin() && !auth.hasPermission([vault.acl()], "read"))
+  ) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_found",
+        message: "The data source you requested was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
+      const viewType = req.query.viewType as ContentNodesViewType;
```

Comment:
`nit` we should do a stricter check here and avoid casting 🙃.

Reply:
it's actually done, the casting is useless :-)